### PR TITLE
Add support for RDP6 32bpp bitmaps

### LINKED ISSUE #109: Codec - RDP6 32bpp
https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpegdi/9b422f69-8e05-4c6d-b6fb-fa02ef75a8f2

[TODO comment in code](https://github.com/Devolutions/IronRDP/blob/ccdd3825b08c64d7736dd3207c3d7281d0445b30/crates/session/src/fast_path.rs#L78)

Implementation should be similar to [RLE](https://github.com/Devolutions/IronRDP/blob/ccdd3825b08c64d7736dd3207c3d7281d0445b30/crates/graphics/src/rle.rs):
- Write to a `&mut [u8]` / `&mut Vec<u8>`
- Read from a `&[u8]`
- Self-contained module

Jira Issue: [ARC-149](https://devolutions.atlassian.net/browse/ARC-149)

[ARC-149]: https://devolutions.atlassian.net/browse/ARC-149?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by pacmancoder**
Added support for RDP6 32bpp bitmaps

### PR scope:
- RDP6 RLE implementation
- `RDP6_BITMAP_STREAM` pdu parsing
- New client params to select color depth and color reduction
- Unit tests and fuzzing for implemented functionality

### New CLI params
To test new functionality, new params `--color-depth` and `--lossy-bitmap-compression` were added.

Test bpp32 mode without lossy compression (RGB  colorspace)
```
cargo run -p ironrdp-client -- -u vmbox -p password --color-depth=32 <IP_ADDRESS>
```

Test bpp32 mode with lossy compression (YCoCg colorspace, subsampling, color loss)
```
cargo run -p ironrdp-client -- -u vmbox -p password --color-depth=32 --lossy-bitmap-compression <IP_ADDRESS>
```

### Preview (32bpp with lossy compression)
изображение

Closes #109

### CODE REVIEW

**Review by CBenoit — COMMENTED** [review]
This looks pretty good, thank you for the high quality PR! :smile:

File: crates/graphics/src/rdp6/mod.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+//! This module provides the RDP6 bitmap decoder implementation
+
+pub(crate) mod bitmap_stream;
+pub(crate) mod rle;
+
+pub use bitmap_stream::BitmapStreamDecoder;
+pub use rle::RleError;
```
[the remark this answers is not in the document]
Comment (pacmancoder) [reply crates/graphics/src/rdp6/mod.rs:7]: Forgot to add `BitmapDecodeError` as reexport, thanks! `decompress_8bpp_plane` on the other hand, should stay private, as rle decoding is only implementation detail of bitmap stream

File: crates/graphics/src/rle.rs [hunk 2]
```diff
@@ -545,30 +544,30 @@ struct Buf<'a> {
...
     fn read_u16(&mut self) -> u16 {
-        u16::from_le_bytes(self.read::<2>())
+        u16::from_le_bytes(self.read_fixed::<2>())
     }
 
     fn read_u24(&mut self) -> u32 {
-        let bytes = self.read::<3>();
+        let bytes = self.read_fixed::<3>();
         u32::from_le_bytes([bytes[0], bytes[1], bytes[2], 0])
     }
```
[the remark this answers is not in the document]
Comment (pacmancoder) [reply crates/graphics/src/rle.rs:572]: Thanks, forgot to change this back after draft implementation, I`ll revert this

File: crates/pdu/src/basic_output/bitmap/rdp6.rs [hunk 3]
```diff
@@ -0,0 +1,257 @@
+use crate::{bitmap::BitmapError, utils::SplitTo};
+use byteorder::{ReadBytesExt, WriteBytesExt};
+use std::io::Write;
```
[the remark this answers is not in the document]
Comment (pacmancoder) [reply crates/pdu/src/basic_output/bitmap/rdp6.rs:3]: Aha, sounds good, I'll switch to `ReadCursor`/`WriteCursor`

**Comment by pacmancoder** [comment]
@CBenoit I updated the code, please review

**Review by CBenoit — APPROVED** [review]
LGTM!

**Pull request merged**